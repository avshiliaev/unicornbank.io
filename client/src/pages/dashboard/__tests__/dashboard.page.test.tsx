import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { cleanup } from '../../../../test-utils';
import DashboardPage from '../dashboard.page';
import { UserState } from '../../../reducers/auth.reducer';
import { shallow } from 'enzyme';

// https://www.robinwieruch.de/react-connected-component-test
// https://hackernoon.com/testing-react-components-with-jest-and-enzyme-41d592c174f

/*
Everything you pass into mockStore will be your Redux store's initial state.
So make sure you provide everything that's needed by your connected React
component to render without any problems.
 */
const mockStore = configureStore([]);
describe('My Connected React-Redux Component', () => {
  afterEach(cleanup);

  it('renders without error', () => {

    const Body = () => {
      return (
        <div>Body</div>
      );
    };

    const auth: UserState = { isLoggedIn: true, userId: '1', userName: 'testUser' };

    const store = mockStore({
      windowSize: {
        greaterThan: {
          extraSmall: false,
          small: false,
          medium: false,
          large: true,
          extraLarge: false,
        },
      },
      auth,
      router: {
        location: {},
      },
    });

    const wrapper = shallow(
      <Provider store={store}>
        <DashboardPage>
          <Body/>
        </DashboardPage>
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();

  });
});


