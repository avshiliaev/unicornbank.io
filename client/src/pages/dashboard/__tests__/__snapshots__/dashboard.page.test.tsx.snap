// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component renders without error 1`] = `ShallowWrapper {}`;
